src/thunks/fetchDataThunk.js

```diff
@@ -0,0 +1,17 @@
+import { batch } from 'react-redux';
+import { loadingEnd, loadingStart } from '../actions/loadingActions';
+import API from '../api';
+
+export function fetchDataThunk(dispatch, category, action) {
```

Comment:
you can dispatch this function instead of creating separate functions, where you call this one and pass a dispatch here